skuber.networking.Ingress fails on Kubernetes 1.18.x

Hi, 

I have encountered two problems with Skuber 2.5.0 when trying to create an ingress object. When I try to create an ingress to an existing service, I get the below exception:

```bash
[ERROR] [08/17/2020 09:53:23.348] [default-akka.actor.default-dispatcher-4] [akka.dispatch.Dispatcher] Failure(skuber.api.client.package$K8SException: Status(v1,Status,ListMeta(,,None),Some(Failure),Some(the API version in the data (networking.k8s.io/v1beta1) does not match the expected API version (extensions/v1beta1)),Some(BadRequest),None,Some(400))) (of class scala.util.Failure)
scala.MatchError: Failure(skuber.api.client.package$K8SException: Status(v1,Status,ListMeta(,,None),Some(Failure),Some(the API version in the data (networking.k8s.io/v1beta1) does not match the expected API version (extensions/v1beta1)),Some(BadRequest),None,Some(400))) (of class scala.util.Failure)
```

I think this is because the companion object in ``skuber.networking.Ingress.scala`` (https://github.com/doriordan/skuber/blob/master/client/src/main/scala/skuber/networking/Ingress.scala#L141) is still using the deprecated api group "extensions/v1beta1" instead of "networking.k8s.io/v1beta1" in the class definition. I have experiment with said solution in my dev environment and it works.

The second problem is that when I try to specify a numeric port in ``Ingress.addHttpRule``, it fails with a validation error from Kubernetes. I think this is because the Kubernetes API for creating ingress expects an integer when the port is supposed to be numeric. If it is a string, it expects the name of a Port resource and not the string value of integer port number.  ``skuber.networking.skuber.Backend`` always expects a string - IMHO it should either accept both int and string or should apply the same validation rule to convert the all numeric port number string to int internally before calling Kubernetes. 

Steps to Reproduce
---------------------------

* Create the hello-minikube service as described here: https://kubernetes.io/docs/setup/learning-environment/minikube/

* Add an entry in ``etc/hosts`` which associates the cluster IP with the hostname "helloservice.acme.com".

* Execute the following:

```scala

import akka.actor.ActorSystem
import akka.stream.ActorMaterializer
import skuber.networking.Ingress
import skuber.api.client.KubernetesClient
import skuber.{ObjectMeta, k8sInit}
import skuber.json.networking.format._
import scala.concurrent.ExecutionContext
import scala.util.Success

object LocalConnect {

  def main(args: Array[String]): Unit = {
    implicit val system: ActorSystem = ActorSystem()
    implicit val ec: ExecutionContext = system.dispatcher
    implicit val materializer: ActorMaterializer = ActorMaterializer()(system)
    val k8s: KubernetesClient = k8sInit

    val metadata: ObjectMeta = ObjectMeta(name="hello-ingress", namespace="default")
    val ingressSpec: Ingress = Ingress(metadata=metadata)
      .addHttpRule(Some("helloservice.acme.com"), Map("/"-> "hello-minikube:8080"))

    k8s.create(ingressSpec).onComplete {
      case Success(_) => println("Successfully created ingress.")
    }
  }
}
```

Thanks in advance! Please let me know if I can provide any other assistance. 

Best,
Dipanjan

--- a/client/src/main/scala/skuber/networking/Ingress.scala
+++ b/client/src/main/scala/skuber/networking/Ingress.scala
@@ -6,6 +6,7 @@ package skuber.networking
 
 import skuber.ResourceSpecification.{Names, Scope}
 import skuber._
+import scala.util.Try
 
 case class Ingress(
   kind: String ="Ingress",
@@ -96,7 +97,7 @@ case class Ingress(
     val paths: List[Ingress.Path] = pathsMap.map { case (path: String, backend: String) =>
       backend match {
         case backendSpec(serviceName, servicePort) =>
-          Ingress.Path(path,Ingress.Backend(serviceName, servicePort))
+          Ingress.Path(path,Ingress.Backend(serviceName, toNameablePort(servicePort)))
         case _ => throw new Exception(s"invalid backend format: expected 'serviceName:servicePort' (got '$backend', for host: $host)")
       }
 
@@ -115,7 +116,7 @@ case class Ingress(
   def withDefaultBackendService(serviceNameAndPort: String): Ingress = {
     serviceNameAndPort match {
       case backendSpec(serviceName, servicePort) =>
-        withDefaultBackendService(serviceName, servicePort)
+        withDefaultBackendService(serviceName, toNameablePort(servicePort))
       case _ => throw new Exception(s"invalid default backend format: expected 'serviceName:servicePort' (got '$serviceNameAndPort')")
     }
   }
@@ -126,7 +127,7 @@ case class Ingress(
    * @param servicePort - service port
    * @return copy of this Ingress with default backend set
    */
-  def withDefaultBackendService(serviceName: String, servicePort: String): Ingress = {
+  def withDefaultBackendService(serviceName: String, servicePort: NameablePort): Ingress = {
     val be = Backend(serviceName, servicePort)
     this.copy(spec=Some(copySpec.copy(backend = Some(be)))
     )
@@ -135,7 +136,8 @@ case class Ingress(
   def addAnnotations(newAnnos: Map[String, String]): Ingress =
     this.copy(metadata = this.metadata.copy(annotations = this.metadata.annotations ++ newAnnos))
 
-
+  private def toNameablePort(port: String): NameablePort =
+    Try(port.toInt).toEither.left.map(_ => port).swap
 }
 
 object Ingress {
@@ -156,7 +158,7 @@ object Ingress {
 
   def apply(name: String) : Ingress = Ingress(metadata=ObjectMeta(name=name))
 
-  case class Backend(serviceName: String, servicePort: String)
+  case class Backend(serviceName: String, servicePort: NameablePort)
   case class Path(path: String, backend: Backend)
   case class HttpRule(paths: List[Path] = List())
   case class Rule(host: Option[String], http: HttpRule)

--- a/client/src/main/scala/skuber/networking/Ingress.scala
+++ b/client/src/main/scala/skuber/networking/Ingress.scala
@@ -141,7 +141,7 @@ case class Ingress(
 object Ingress {
 
   val specification: NonCoreResourceSpecification = NonCoreResourceSpecification(
-    apiGroup = "extensions",
+    apiGroup = "networking.k8s.io",
     version = "v1beta1",
     scope = Scope.Namespaced,
     names = Names(

--- a/client/src/test/scala/skuber/network/IngressSpec.scala
+++ b/client/src/test/scala/skuber/network/IngressSpec.scala
@@ -52,6 +52,12 @@ class IngressSpec extends Specification {
         |                "serviceName": "service",
         |                "servicePort": "http"
         |              }
+        |            },
+        |            {
+        |              "backend": {
+        |                "serviceName": "ssh",
+        |                "servicePort": 22
+        |              }
         |            }
         |          ]
         |        }
@@ -79,10 +85,10 @@ class IngressSpec extends Specification {
     ing.name mustEqual "example-ingress"
 
     ing.spec.get.rules.head.host must beSome("example.com")
-    ing.spec.get.rules.head.http.paths must_== List(Ingress.Path(
-      path = "",
-      backend = Ingress.Backend("service", "http")
-    ))
+    ing.spec.get.rules.head.http.paths must_== List(
+      Ingress.Path(path = "", backend = Ingress.Backend("service", "http")),
+      Ingress.Path(path = "", backend = Ingress.Backend("ssh", 22))
+    )
     ing.spec.get.tls must_== List(Ingress.TLS(
       hosts = List("abc","def"),
       secretName = None

